### PR TITLE
Fix Mac and laptop mouse controls

### LINKED ISSUE #285: Picking segfault on line mesh
Hi,
I haven't tested a lot, but when I add a RM_LINES render object, it raise an OpenGL invalid operation 
```c++
meshBezier->setRenderMode( Engine::Mesh::RM_LINES );
/* load geometry and set config */
    Engine::RenderObject *ro =
        new Engine::RenderObject( "bezierLine", tip, Engine::RenderObjectType::Fancy );
    ro->setMesh( meshBezier );
    Ra::Engine::RadiumEngine::getInstance()->getRenderObjectManager()->addRenderObject( ro );
```
Relevant trace:
```bash
- 12:34:54 ERROR: OpenGL error (/media/jupi/Repositories/painty/external/Radium-Engine/src/Engine/Renderer/Mesh/Mesh.cpp:49, glDrawElements( static_cast<GLenum >(m_renderMode), m_numElements, GL_UNSIGNED_INT, (void*)0 )) :  Invalid operation : The specified operation is not allowed in the current state. The offending command is ignored and has no other side effect than to set the error flag.
(GL_INVALID_OPERATION : 0xGL_INVALID_OPERATION).

Thread 1 "painty" received signal SIGTRAP, Trace/breakpoint trap.
bt
Ra::Engine::Mesh::render (this=0x555555faac60)
    at /media/jupi/Repositories/painty/external/Radium-Engine/src/Engine/Renderer/Mesh/Mesh.cpp:51
51	        }
(gdb) bt
#0  0x00007ffff70e1374 in Ra::Engine::Mesh::render() (this=0x555555faac60)
    at /media/jupi/Repositories/painty/external/Radium-Engine/src/Engine/Renderer/Mesh/Mesh.cpp:51
#1  0x00007ffff71fbda7 in Ra::Engine::Renderer::renderForPicking(Ra::Engine::RenderData const&, std::array<Ra::Engine::ShaderProgram const*, 3ul> const&, std::array<std::vector<std::shared_ptr<Ra::Engine::RenderObject>, std::allocator<std::shared_ptr<Ra::Engine::RenderObject> > >, 3ul> const&) (this=0x555557c454c0, renderData=..., pickingShaders=..., renderQueuePicking=...)
    at /media/jupi/Repositories/painty/external/Radium-Engine/src/Engine/Renderer/Renderer.cpp:346
#2  0x00007ffff71fccd1 in Ra::Engine::Renderer::doPicking(Ra::Engine::RenderData const&) (this=0x555557c454c0, renderData=...)
    at /media/jupi/Repositories/painty/external/Radium-Engine/src/Engine/Renderer/Renderer.cpp:380
#3  0x00007ffff71faf2a in Ra::Engine::Renderer::render(Ra::Engine::RenderData const&) (this=0x555557c454c0, data=...)
    at /media/jupi/Repositories/painty/external/Radium-Engine/src/Engine/Renderer/Renderer.cpp:160
#4  0x00007ffff64622f9 in Ra::Gui::Viewer::startRendering(float) (this=
    0x5555565dfa90, dt=0.0166666675)
    at /media/jupi/Repositories/painty/external/Radium-Engine/src/GuiBase/Viewer/Viewer.cpp:533
```

### PR DESCRIPTION
**Pull request by hoshiryu**
Add possibility to have modifiers on mouse buttons ;
Adapt Viewer's mousePressEvent and mouseMoveEvent accordingly ;
Fix alternative keymapping config to work on computers with pad only (e.g. mac and laptop) ;
Add a warning if an action key is bound to multiple actions ;

Add transforms management to MeshFeatureTracking ;
Add picking support for some OpenGL primitive types (see Mesh.cpp for supported types ) ;

Fixes #285

**Note**: the default configuration uses the same key for `GIZMOMANAGER_MANIPULATION`, `VIEWER_BUTTON_CAST_RAY_QUERY` and `COLORWIDGET_PRESSBUTTON` ; while the alternative configuration uses the same key for `VIEWER_BUTTON_PICKING_QUERY` and `VIEWER_BUTTON_CAST_RAY_QUERY`, and the same key for `TRACKBALLCAMERA_MANIPULATION` and `COLORWIDGET_PRESSBUTTON`.
This is currently not a big problem since the `VIEWER_BUTTON_CAST_RAY_QUERY` action is activated only if `VIEWER_RAYCAST_QUERY` is active, which overrides the other actions, and the `COLORWIDGET_PRESSBUTTON` is currently not in use.

### CODE REVIEW

File: src/GuiBase/Viewer/Viewer.cpp [hunk 1]
```diff
@@ -300,8 +300,13 @@ namespace Ra
                 e->rayCastQuery(r);
             }
         }
+        else if ( keyMap->actionTriggered( event, Gui::KeyMappingManager::TRACKBALLCAMERA_MANIPULATION ) )
+        {
+            m_camera->handleMousePressEvent(event);
+        }
         else if ( keyMap->actionTriggered( event, Gui::KeyMappingManager::GIZMOMANAGER_MANIPULATION ) )
         {
+            std::cout << "gizmo" << std::endl;
```
Comment (dlyr) [inline src/GuiBase/Viewer/Viewer.cpp]: Please remove these cout ...

File: src/GuiBase/Viewer/Viewer.cpp [hunk 2]
```diff
@@ -310,12 +315,9 @@ namespace Ra
                 m_gizmoManager->handleMousePressEvent(event);
             }
         }
-        else if ( keyMap->actionTriggered( event, Gui::KeyMappingManager::TRACKBALLCAMERA_MANIPULATION ) )
-        {
-            m_camera->handleMousePressEvent(event);
-        }
         else if ( keyMap->actionTriggered( event, Gui::KeyMappingManager::VIEWER_BUTTON_PICKING_QUERY ) )
         {
+            std::cout << "picking" << std::endl;
```
Comment (dlyr) [inline src/GuiBase/Viewer/Viewer.cpp]: please remove

**Comment by hoshiryu** [comment]
Still issues on mac...

**Comment by nmellado** [comment]
@hoshiryu , can you please update from Master, so we can validate the PR with CI builds ?

**Comment by hoshiryu** [comment]
This PR is not ready yet

**Comment by nmellado** [comment]
Ok.
Please remove the WIP label when you are done.

**Comment by hoshiryu** [comment]
PR ready for review

**Comment by nmellado** [comment]
Just to be sure: as far as I see, this fixes issue #285, right @hoshiryu ?

**Comment by hoshiryu** [comment]
Yes, see edited first comment

**Comment by nmellado** [comment]
First comment says Not Ready Yet ;)

**Comment by MathiasPaulin** [comment]
Well, I havent time yet to fix the picking rendering problem but, as mentioned in the tittle of this PR. It is just for mouse control.
So, yep, I agree to make the final review of this PR and I'll open an issue for picking not functionning ...

**Pull request merged**